src/meltano/core/logging/formatters.py

```diff
@@ -26,23 +26,52 @@ def _process_formatter(processor: Processor):
 
     Args:
         processor: A structlog message processor such as structlog.dev.ConsoleRenderer.
+
+    Returns:
+        A configured log processor.
     """
     return structlog.stdlib.ProcessorFormatter(
         processor=processor, foreign_pre_chain=LEVELED_TIMESTAMPED_PRE_CHAIN
     )
 
 
-def console_log_formatter(colors: bool = False) -> None:
-    """Create a logging formatter for console rendering that supports colorization."""
-    return _process_formatter(structlog.dev.ConsoleRenderer(colors=colors))
+def console_log_formatter(
+    colors: bool = False, exception_formatter: str = "plain"
+) -> None:
+    """Create a logging formatter for console rendering that supports colorization.
+
+    Args:
+        colors: Add color to output.
+        exception_formatter: Format exceptions as "plain" or "rich".
+
+    Returns:
+        A configured console log formatter.
+    """
+    processor = (
+        structlog.dev.ConsoleRenderer(colors=colors)
+        if exception_formatter == "rich"
+        else structlog.dev.ConsoleRenderer(
+            colors=colors, exception_formatter=structlog.dev.plain_traceback
+        )
+    )
+    return _process_formatter(processor)
```

Comment:
Perhaps this is more readable and makes it easier to pass more parameters to `ConsoleRenderer` in the future:

```suggestion
    exception_formatter = (
        structlog.dev.rich_traceback
        if exception_formatter = "rich"
        else structlog.dev.plain_traceback
    )
    return structlog.dev.ConsoleRenderer(
        colors=colors,
        exception_formatter=exception_formatter,
    )
```